### ISSUE TITLE
Monitoring of Backends

### ISSUE THREAD

**Issue by pyte1** [strukturag/nextcloud-spreed-signaling#182]
Hi,

i trying to find a way to monitor the connections/users on each backend that i defined in the servers.conf. I tried to do it with some scripting and analysing the output but with no great luck. Is there any easy way to messure the current connection/user count for a specific backend or all backends overall?

**Comment by pyte1** [comment]
Thank you for your reply. 

Is there any configuration needed for this to work? As it does not seem to work OOB

**Comment by fancycode** [comment]
This should be working out-of-the box with 0.4.0 and newer for clients connecting from whitelisted IP addresses (`127.0.0.1` by default):
https://github.com/strukturag/nextcloud-spreed-signaling/blob/414f78dd87030405aeef0a6c5e612437f7054797/server.conf.in#L226

**Comment by fancycode** [comment]
Example from my local dev setup:
```
[...]
# HELP signaling_hub_rooms The current number of rooms per backend
# TYPE signaling_hub_rooms gauge
signaling_hub_rooms{backend="compat"} 1
# HELP signaling_hub_sessions The current number of sessions per backend
# TYPE signaling_hub_sessions gauge
signaling_hub_sessions{backend="compat",clienttype="client"} 1
[...]
```

**Comment by pyte1** [comment]
Yep i've got it.  I guess "signaling_hub_rooms" and "signaling_hub_sessions" gets only displayed if there are any? 

Right now i only see:

`# HELP signaling_backend_current The current number of configured backends
# TYPE signaling_backend_current gauge
signaling_backend_current 1
# HELP signaling_hub_sessions_resume_failed_total The total number of failed session resume requests
# TYPE signaling_hub_sessions_resume_failed_total counter
signaling_hub_sessions_resume_failed_total 0`

But there is no one using it right now.

**Comment by fancycode** [comment]
Yeah, some values only get created once the corresponding entry is used, i.e. a backend gets sessions / rooms or a room has sessions, e.g.:
```
# HELP signaling_room_sessions The current number of sessions in a room
# TYPE signaling_room_sessions gauge
signaling_room_sessions{backend="compat",clienttype="client",room="abcdefgh"} 1
```

This is because the metrics subsystem doesn't know about all possible values and it doesn't make sense to have all potential values in the output all the time, e.g. the sessions per room as shown above. Once the number of sessions in a room reach zero, the value is no longer returned.

**Comment by cari-coders** [comment]
> The signaling server (and -proxy) have a `/metrics` endpoint that provide data that can be imported in Prometheus / Grafana. Unfortunately this is still lacking documentation but you should get some data by loading `http://<signaling-ip>:<signaling-port>/metrics`.

I am getting an "authentication check failed" reply. Which headers must i include in the request? And can I find documentation about it somewhere?

**Comment by fancycode** [comment]
The metrics are documented here:
https://github.com/strukturag/nextcloud-spreed-signaling/blob/master/docs/prometheus-metrics.md

Authentication is IP-based and must be configured in the `server.conf`:
https://github.com/strukturag/nextcloud-spreed-signaling/blob/8a53bab9cf48f5aafb54084bc1055e30353bf726/server.conf.in#L264-L267
